tests/cpp-runtime/hexagon/hexagon_thread_manager_tests.cc

```diff
@@ -54,6 +55,15 @@ TEST_F(HexagonThreadManagerTest, ctor_errors) {
   ASSERT_THROW(HexagonThreadManager(6, stack_size, 9), InternalError);
   // pipe too big
   ASSERT_THROW(HexagonThreadManager(6, stack_size, 0x10000000), InternalError);
+  // hw resources count doesn't match thread count
+  ASSERT_THROW(HexagonThreadManager(6, stack_size, pipe_size, {DMA_0}), InternalError);
+  // hw resources doesn't match specific supported configuration
```

Comment:
We should come back and revisit this.  Having this _very_ strict specification of hardware resources is probably not what we want long term.  Let's see if we can make this a little more flexible going forward.  That said, I am OK with this PR as an interim step.